File: backend/src/main/java/com/cdac/farmermarketplace/service/NotificationService.java
package com.cdac.farmermarketplace.service;

import java.util.List;

import com.cdac.farmermarketplace.dto.response.NotificationResponseDto;
import com.cdac.farmermarketplace.entity.User;

public interface NotificationService {

    // Automatic notification creation
    void createNotification(
            User user,
            String title,
            String message
    );

    List<NotificationResponseDto> getUserNotifications(Long userId);

    List<NotificationResponseDto> getUnreadNotifications(Long userId);

    Long getUnreadCount(Long userId);

    NotificationResponseDto markAsRead(Long notificationId);

    void deleteNotification(Long notificationId);
    void markAllAsRead(Long userId);
}

File: frontend/src/services/notificationService.js

import api from "./api";

// ================= GET MY NOTIFICATIONS =================

export const getMyNotifications = async (userId) => {
  const response = await api.get(
    `/notifications/user/${userId}`
  );

  return response.data;
};

// ================= GET MY UNREAD COUNT =================

export const getUnreadCount = async (userId) => {
  const response = await api.get(
    `/notifications/user/${userId}/count`
  );

  return response.data;
};

// ================= MARK AS READ =================

export const markAsRead = async (id) => {
  const response = await api.put(
    `/notifications/${id}/read`
  );

  return response.data;
};

// ================= MARK ALL AS READ =================

export const markAllAsRead = async (userId) => {
  const response = await api.put(
    `/notifications/user/${userId}/read-all`
  );

  return response.data;
};
export const getUnreadNotifications = async (userId) => {
  const response = await api.get(
    `/notifications/user/${userId}/unread`
  );

  return response.data;
};